### PR TITLE
[profiler] Split up profiler test file

### PR DESCRIPTION
**Pull request by briancoutinho**
To help with issues on test time out split profiler test file into 4 files.
- profiler
- record_function
- execution_trace
- torch_tidy

cc @albanD

### CODE REVIEW

**Comment by briancoutinho** [comment]
@albanD This is failing with PR size restriction but that is because i am splitting up a large file already, can you grant me an exception cc @aaronenyeshi , @davidberard98

**Review by albanD — COMMENTED** [review]
Yes this kind of exception are all good as long as the real reviewer is happy with it!

File: test/profiler/test_execution_trace.py [hunk 1]
```diff
@@ -0,0 +1,374 @@
...
+# in that test with their tids. The events that correspond to these lingering
+# threads all have TID of (uint64_t)(-1) which is invalid.
+# The work around is turnning off monitoring thread when tqdm is loaded.
+# Since these are unit tests, it is safe to turn off monitor thread.
+try:
+    import tqdm
+
+    tqdm.tqdm.monitor_interval = 0
+except ImportError:
+    None
```
Comment (shengfukevin) [inline test/profiler/test_execution_trace.py:14]: Please change None to Pass. The same for other three files.

**Review by shengfukevin — APPROVED** [review]
LGTM! One minor comments.

File: test/profiler/test_execution_trace.py (hunk 1 above)
Comment (briancoutinho) [reply test/profiler/test_execution_trace.py:14]: ohh okk

**Comment by briancoutinho** [comment]
> > By the way, in capturing triton kernel PR, I changed the test test_execution_trace_with_pt2, please copy it over. Thanks
> 
> I think that PR got reverted. Can we keep the most recent test_execution_trace_with_pt2, and then land the changes for test_execution_trace_with_pt2 in another PR?

oh no :(  okk let me do that

**Review by aaronenyeshi — CHANGES_REQUESTED** [review]
This is great! Thanks for splitting up the test_profiler.py, this is a great idea as it was getting too long and taking a long time to run as well. I have a few small questions.

File: test/profiler/test_profiler.py [hunk 2]
```diff
@@ -2116,18 +1639,6 @@ def test_event_list(self):
...
-            return node
-
-
-def find_node_with_regex(nodes, pattern):
-    for node in _utils.traverse_dfs(nodes):
-        if re.search(pattern, node.name):
-            return node
-
-
 class SimpleNet(nn.Module):
```
[the remark this answers is not in the document]
Comment (briancoutinho) [reply test/profiler/test_profiler.py:1646]: ohh SimpleNet is also used test_kineto_profiler_multiple_steppers() I made a copy of this class in torch_tidy test file

**Comment by shengfukevin** [comment]
> > > By the way, in capturing triton kernel PR, I changed the test test_execution_trace_with_pt2, please copy it over. Thanks
> > 
> > 
> > I think that PR got reverted. Can we keep the most recent test_execution_trace_with_pt2, and then land the changes for test_execution_trace_with_pt2 in another PR?
> 
> oh no :( okk let me do that

Jean Schmidt is helping remerging https://github.com/pytorch/pytorch/pull/124775, is it ok to get my PR merge first?

**Review by aaronenyeshi — APPROVED** [review]
LGTM!

**Pull request closed without merging**